### DISCUSSION TITLE
Access Token in React.useEffect generally works, but comes back undefined when page is refreshed / reloaded

### DISCUSSION THREAD

**Discussion by devrosada** [iamrosada/friendly-venus#4] [Q&A] [answered]
I have a simple React page / component in Gatsby that makes an API call. For this API call I need a token. I use [gatsby-theme-auth0](https://www.gatsbyjs.com/plugins/gatsby-theme-auth0/) to obtain this token via their AuthService object.

I am starting the API call in my useEffect. It looks like this:

`
useEffect(() => {

    async function fetchFromAPI() {
      try {
        const data = await fetchData()
        setData(data)
      }
    }

    
    fetchFromAPI()
  }, [])
`

**Comment by iamrosada** [comment, marked as answer]
I don't know if you have the authentication in a hook already or not, but you need to check if the user is authenticated before you make any api call, especially those that on app init. Do you have a hook/context when you handle the authentication ? 

const {isAuthenticated} = useContext(userAuthenticatedContext)
`
 useEffect(() => {

    async function fetchFromAPI() {
      try {
        const data = await fetchData()
        setData(data)
      }
    }
    
    if(isAuthenticated) fetchFromAPI()
  }, [isAuthenticated])

`
